process-analyser/src_hd/com/axonivy/utils/process/analyser/component/CustomFilter/CustomFilter.xhtml

```diff
@@ -4,89 +4,94 @@
             update="custom-filter-group filter-options-group" />
           <p:ajax event="itemUnselect" update="custom-filter-group filter-options-group" process="@this"
             partialSubmit="true" />
-          <f:selectItems value="#{processesAnalyticsBean.customFieldsByType.keySet()}" var="customFieldFilter"
+          <f:selectItems value="#{managedBean.customFieldsByType.keySet()}" var="customFieldFilter"
             itemValue="#{customFieldFilter.customFieldMeta.name()}"
             itemLabel="#{customFieldFilter.customFieldMeta.name()}" />
         </p:selectCheckboxMenu>
       </div>
     </h:panelGroup>
 
     <h:panelGroup layout="block" styleClass="field grid mb-0" id="filter-options-group">
-      <ui:repeat value="#{processesAnalyticsBean.selectedCustomFilters.entrySet()}" var="fieldEntry">
-        <h:panelGroup styleClass="ui-fluid col-4" layout="block" rendered="#{processesAnalyticsBean.filterDropdownVisible}">
+      <ui:repeat value="#{managedBean.selectedCustomFilters.entrySet()}" var="fieldEntry">
+        <h:panelGroup styleClass="ui-fluid col-4" layout="block" rendered="#{managedBean.filterDropdownVisible}">
           <div class="col-4 p-0 m-0 mb-2">
             <p:outputLabel value="#{fieldEntry.key.customFieldMeta.name()}"
-              rendered="#{processesAnalyticsBean.filterDropdownVisible}" />
+              rendered="#{managedBean.filterDropdownVisible}" />
           </div>
 
           <!-- NUMBER -->
-          <h:outputText id="number-range"
-            rendered="#{processesAnalyticsBean.filterDropdownVisible and fieldEntry.key.customFieldMeta.type() eq 'NUMBER'}"
-            value="Between #{processesAnalyticsBean.selectedCustomFilters[fieldEntry.key][0]} and #{processesAnalyticsBean.selectedCustomFilters[fieldEntry.key][1]}" />
+          <h:outputText id="number-range" rendered="#{fieldEntry.key.customFieldMeta.type() eq 'NUMBER'}"
+            value="#{managedBean.getRangeDisplayForNumberType(managedBean.selectedCustomFilters[fieldEntry.key])}" />
 
-          <p:slider for="min-value, max-value" range="true" display="number-range"
+          <p:slider for="
+            min-value, max-value" range="true" display="number-range"
             displayTemplate="Between {min} and {max}" styleClass="m-2"
-            minValue="#{processesAnalyticsBean.getMinValue(fieldEntry.key.customFieldMeta.name())}"
-            maxValue="#{processesAnalyticsBean.getMaxValue(fieldEntry.key.customFieldMeta.name())}"
-            rendered="#{processesAnalyticsBean.filterDropdownVisible and fieldEntry.key.customFieldMeta.type() eq 'NUMBER'}">
+            minValue="#{managedBean.getMinValue(fieldEntry.key.customFieldMeta.name())}"
+            maxValue="#{managedBean.getMaxValue(fieldEntry.key.customFieldMeta.name())}"
+            rendered="#{managedBean.filterDropdownVisible and 'NUMBER' == fieldEntry.key.customFieldMeta.type()}">
```

Comment:
we are cloning this condition a lot, I think we can group the code from line 45-60 inside a <h:panelgroup> tag, then apply the render condition for it